parley/src/layout/data.rs

```diff
@@ -287,176 +352,118 @@ impl<B: Brush> LayoutData<B> {
             bidi_level,
         });
     }
-
-    #[allow(unused_assignments)]
     #[allow(clippy::too_many_arguments)]
     pub(crate) fn push_run(
         &mut self,
         font: Font,
         font_size: f32,
-        synthesis: Synthesis,
-        shaper: Shaper<'_>,
+        synthesis: fontique::Synthesis,
+        glyph_buffer: &harfrust::GlyphBuffer,
         bidi_level: u8,
         word_spacing: f32,
         letter_spacing: f32,
+        source_text: &str,
+        char_infos: &[(swash::text::cluster::CharInfo, u16)], // From text analysis
+        text_range: Range<usize>,                             // The text range this run covers
+        variations: impl Iterator<Item = harfrust::Variation>,
     ) {
+        let coords_start = self.coords.len();
+        let mut variations = variations.peekable();
+        if variations.peek().is_some() {
+            self.store_variations(&font, variations);
+        }
```

Comment:
You should just be able to grab the final set of normalized coordinates from [`ShaperInstance::coords`](https://docs.rs/harfrust/latest/harfrust/struct.ShaperInstance.html#method.coords) and avoid reimplementing all the normalization logic :)

Reply:
So much better!!! Thank you - implemented in https://github.com/linebender/parley/pull/400/commits/b8c1b51e8683e96f30c55a12f74ec7be3432be71